spikeextractors/extractors/intanrecordingextractor/intanrecordingextractor.py

```diff
@@ -52,8 +52,15 @@ def get_sampling_frequency(self):
     def get_traces(self, channel_ids=None, start_frame=None, end_frame=None):
         channel_idxs = np.array([self._channel_ids.index(ch) for ch in channel_ids])
         analog_chans = self._analog_channels[channel_idxs]
-        return self._recording._read_analog(channels=analog_chans,
-                                            i_start=start_frame, i_stop=end_frame)[:, channel_idxs].T
+        tr = self._recording._read_analog(
+            channels=analog_chans,
+            i_start=start_frame,
+            i_stop=end_frame
+        )
+        if len(channel_idxs) == 1:
+            return tr.T
+        else:
+            return tr[:, channel_idxs].T
```

Comment:
Actually, why is the secondary indexing even needed? Shouldn't `analog_chans` already have the correct indexing behavior?
```suggestion
        return self._recording._read_analog(channels=analog_chans,
                                            i_start=start_frame, i_stop=end_frame).T
```

Reply:
Hi Cody! I made some changes recently and might have broken something. Thanks for the fix! I'll test it tomorrow and merge :)